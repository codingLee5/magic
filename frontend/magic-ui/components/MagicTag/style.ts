import { createStyles } from "antd-style"

export const useStyles = createStyles(({ css, prefixCls, token }) => ({
	tag: css`
		--${prefixCls}-tag-default-bg: ${token.magicColorUsages.fill[0]} !important;
		--${prefixCls}-color-border: transparent !important;
		--${prefixCls}-border-radius-sm: 8px;
		padding-inline: 4px;
		padding-block: 4px;
		display: flex;
		align-items: center;
		justify-content: center;

		> div {
			gap: 4px !important;

		}
	`,
}))
