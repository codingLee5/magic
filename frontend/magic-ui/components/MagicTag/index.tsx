import { Tag, type TagProps } from "antd"
import MagicIcon from "../MagicIcon"
import { IconX } from "@tabler/icons-react"
import { useStyles } from "./style"

export type MagicTagProps = TagProps

function MagicTag({ className, ...props }: MagicTagProps) {
	const { styles, cx } = useStyles()

	return (
		<Tag
			className={cx(styles.tag, className)}
			closeIcon={<MagicIcon component={IconX} size={16} stroke={2} />}
			{...props}
		/>
	)
}

export default MagicTag
