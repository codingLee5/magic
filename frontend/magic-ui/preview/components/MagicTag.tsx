import React from "react"
import { Space } from "antd"
import MagicTag from "../../components/MagicTag"
import ComponentDemo from "./Container"

const TagDemo: React.FC = () => {
	return (
		<div>
			<ComponentDemo
				title="基础标签"
				description="最基本的标签组件"
				code="<MagicTag>默认标签</MagicTag>"
			>
				<Space wrap>
					<MagicTag>默认标签</MagicTag>
					<MagicTag color="blue">蓝色标签</MagicTag>
					<MagicTag color="green">绿色标签</MagicTag>
					<MagicTag color="red">红色标签</MagicTag>
				</Space>
			</ComponentDemo>

			<ComponentDemo
				title="标签颜色"
				description="支持多种预设颜色"
				code="color: 'blue' | 'green' | 'red' | 'orange' | 'purple'"
			>
				<Space wrap>
					<MagicTag color="blue">蓝色</MagicTag>
					<MagicTag color="green">绿色</MagicTag>
					<MagicTag color="red">红色</MagicTag>
					<MagicTag color="orange">橙色</MagicTag>
					<MagicTag color="purple">紫色</MagicTag>
				</Space>
			</ComponentDemo>

			<ComponentDemo title="自定义样式" description="通过style自定义样式" code="style">
				<Space wrap>
					<MagicTag style={{ backgroundColor: "#f0f0f0", color: "#333" }}>
						自定义样式
					</MagicTag>
					<MagicTag
						style={{
							background: "linear-gradient(45deg, #ff6b6b, #4ecdc4)",
							color: "white",
						}}
					>
						渐变背景
					</MagicTag>
				</Space>
			</ComponentDemo>
		</div>
	)
}

export default TagDemo
