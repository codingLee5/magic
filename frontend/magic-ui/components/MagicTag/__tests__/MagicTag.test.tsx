import { render, screen } from "@testing-library/react"
import MagicThemeProvider from "../../ThemeProvider"
import { describe, it, expect } from "vitest"
import MagicTag from "../index"

const renderWithTheme = (component: React.ReactElement) =>
	render(<MagicThemeProvider theme="light">{component}</MagicThemeProvider>)

describe("MagicTag", () => {
	it("应该正常渲染", () => {
		renderWithTheme(<MagicTag>标签内容</MagicTag>)
		expect(screen.getByText("标签内容")).toBeInTheDocument()
	})

	// 快照测试
	describe("快照测试", () => {
		it("默认标签快照", () => {
			const { asFragment } = renderWithTheme(<MagicTag>默认标签</MagicTag>)
			expect(asFragment()).toMatchSnapshot()
		})

		it("不同颜色标签快照", () => {
			const { asFragment: blueFragment } = renderWithTheme(
				<MagicTag color="blue">蓝色标签</MagicTag>,
			)
			expect(blueFragment()).toMatchSnapshot()

			const { asFragment: redFragment } = renderWithTheme(
				<MagicTag color="red">红色标签</MagicTag>,
			)
			expect(redFragment()).toMatchSnapshot()

			const { asFragment: greenFragment } = renderWithTheme(
				<MagicTag color="green">绿色标签</MagicTag>,
			)
			expect(greenFragment()).toMatchSnapshot()
		})

		it("可关闭标签快照", () => {
			const { asFragment } = renderWithTheme(<MagicTag closable>可关闭标签</MagicTag>)
			expect(asFragment()).toMatchSnapshot()
		})

		it("不同样式标签快照", () => {
			const { asFragment: successFragment } = renderWithTheme(
				<MagicTag color="success">成功标签</MagicTag>,
			)
			expect(successFragment()).toMatchSnapshot()

			const { asFragment: warningFragment } = renderWithTheme(
				<MagicTag color="warning">警告标签</MagicTag>,
			)
			expect(warningFragment()).toMatchSnapshot()
		})

		it("边框样式标签快照", () => {
			const { asFragment } = renderWithTheme(<MagicTag bordered>边框标签</MagicTag>)
			expect(asFragment()).toMatchSnapshot()
		})

		it("图标标签快照", () => {
			const { asFragment } = renderWithTheme(
				<MagicTag icon={<span>🏷️</span>}>带图标标签</MagicTag>,
			)
			expect(asFragment()).toMatchSnapshot()
		})
	})
})
