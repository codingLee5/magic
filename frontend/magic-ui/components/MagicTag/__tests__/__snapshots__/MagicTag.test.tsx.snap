// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MagicTag > х┐лчЕзц╡ЛшпХ > ф╕НхРМца╖х╝ПцаЗчн╛х┐лчЕз 1`] = `
<DocumentFragment>
  <span
    class="magic-tag magic-tag-success acss-1mtoqz4 css-dev-only-do-not-override-yotw4n css-var-r6"
  >
    цИРхКЯцаЗчн╛
    <svg
      aria-label="Close"
      class="tabler-icon tabler-icon-x magic-tag-close-icon"
      fill="none"
      height="16"
      stroke="rgba(28,29,35, 0.8)"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M18 6l-12 12"
      />
      <path
        d="M6 6l12 12"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`MagicTag > х┐лчЕзц╡ЛшпХ > ф╕НхРМца╖х╝ПцаЗчн╛х┐лчЕз 2`] = `
<DocumentFragment>
  <span
    class="magic-tag magic-tag-warning acss-1mtoqz4 css-dev-only-do-not-override-yotw4n css-var-r7"
  >
    шнжхСКцаЗчн╛
    <svg
      aria-label="Close"
      class="tabler-icon tabler-icon-x magic-tag-close-icon"
      fill="none"
      height="16"
      stroke="rgba(28,29,35, 0.8)"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M18 6l-12 12"
      />
      <path
        d="M6 6l12 12"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`MagicTag > х┐лчЕзц╡ЛшпХ > ф╕НхРМщвЬшЙ▓цаЗчн╛х┐лчЕз 1`] = `
<DocumentFragment>
  <span
    class="magic-tag magic-tag-blue acss-1mtoqz4 css-dev-only-do-not-override-yotw4n css-var-r2"
  >
    шУЭшЙ▓цаЗчн╛
    <svg
      aria-label="Close"
      class="tabler-icon tabler-icon-x magic-tag-close-icon"
      fill="none"
      height="16"
      stroke="rgba(28,29,35, 0.8)"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M18 6l-12 12"
      />
      <path
        d="M6 6l12 12"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`MagicTag > х┐лчЕзц╡ЛшпХ > ф╕НхРМщвЬшЙ▓цаЗчн╛х┐лчЕз 2`] = `
<DocumentFragment>
  <span
    class="magic-tag magic-tag-red acss-1mtoqz4 css-dev-only-do-not-override-yotw4n css-var-r3"
  >
    ч║вшЙ▓цаЗчн╛
    <svg
      aria-label="Close"
      class="tabler-icon tabler-icon-x magic-tag-close-icon"
      fill="none"
      height="16"
      stroke="rgba(28,29,35, 0.8)"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M18 6l-12 12"
      />
      <path
        d="M6 6l12 12"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`MagicTag > х┐лчЕзц╡ЛшпХ > ф╕НхРМщвЬшЙ▓цаЗчн╛х┐лчЕз 3`] = `
<DocumentFragment>
  <span
    class="magic-tag magic-tag-green acss-1mtoqz4 css-dev-only-do-not-override-yotw4n css-var-r4"
  >
    ч╗┐шЙ▓цаЗчн╛
    <svg
      aria-label="Close"
      class="tabler-icon tabler-icon-x magic-tag-close-icon"
      fill="none"
      height="16"
      stroke="rgba(28,29,35, 0.8)"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M18 6l-12 12"
      />
      <path
        d="M6 6l12 12"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`MagicTag > х┐лчЕзц╡ЛшпХ > хПпхЕ│щЧнцаЗчн╛х┐лчЕз 1`] = `
<DocumentFragment>
  <span
    class="magic-tag acss-1mtoqz4 css-dev-only-do-not-override-yotw4n css-var-r5"
  >
    хПпхЕ│щЧнцаЗчн╛
    <svg
      aria-label="Close"
      class="tabler-icon tabler-icon-x magic-tag-close-icon"
      fill="none"
      height="16"
      stroke="rgba(28,29,35, 0.8)"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M18 6l-12 12"
      />
      <path
        d="M6 6l12 12"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`MagicTag > х┐лчЕзц╡ЛшпХ > хЫ╛цаЗцаЗчн╛х┐лчЕз 1`] = `
<DocumentFragment>
  <span
    class="magic-tag acss-1mtoqz4 css-dev-only-do-not-override-yotw4n css-var-r9"
  >
    <span>
      ЁЯП╖я╕П
    </span>
    <span>
      х╕жхЫ╛цаЗцаЗчн╛
    </span>
    <svg
      aria-label="Close"
      class="tabler-icon tabler-icon-x magic-tag-close-icon"
      fill="none"
      height="16"
      stroke="rgba(28,29,35, 0.8)"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M18 6l-12 12"
      />
      <path
        d="M6 6l12 12"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`MagicTag > х┐лчЕзц╡ЛшпХ > ш╛╣цбЖца╖х╝ПцаЗчн╛х┐лчЕз 1`] = `
<DocumentFragment>
  <span
    class="magic-tag acss-1mtoqz4 css-dev-only-do-not-override-yotw4n css-var-r8"
  >
    ш╛╣цбЖцаЗчн╛
    <svg
      aria-label="Close"
      class="tabler-icon tabler-icon-x magic-tag-close-icon"
      fill="none"
      height="16"
      stroke="rgba(28,29,35, 0.8)"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M18 6l-12 12"
      />
      <path
        d="M6 6l12 12"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`MagicTag > х┐лчЕзц╡ЛшпХ > щ╗ШшодцаЗчн╛х┐лчЕз 1`] = `
<DocumentFragment>
  <span
    class="magic-tag acss-1mtoqz4 css-dev-only-do-not-override-yotw4n css-var-r1"
  >
    щ╗ШшодцаЗчн╛
    <svg
      aria-label="Close"
      class="tabler-icon tabler-icon-x magic-tag-close-icon"
      fill="none"
      height="16"
      stroke="rgba(28,29,35, 0.8)"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M18 6l-12 12"
      />
      <path
        d="M6 6l12 12"
      />
    </svg>
  </span>
</DocumentFragment>
`;
